packages/next/src/server/base-http/web.ts

```diff
@@ -99,12 +87,25 @@ export class WebNextResponse extends BaseNextResponse<WritableStream> {
     return this
   }
 
-  send() {
-    this.sendResolve?.()
+  private readonly sendPromise = Promise.withResolvers<void>()
+  private _sent = false
+  public send() {
+    this.sendPromise.resolve()
     this._sent = true
   }
 
-  toResponse() {
-    return this.response
+  get sent() {
+    return this._sent
+  }
+
+  public async toResponse() {
+    // If we haven't called `send` yet, wait for it to be called.
+    if (!this.sent) await this.sendPromise.promise
```

Comment:
`toResponse` is (as expected) called every request. As soon as you call `await` it will pause execution to get the value of the promise (even if it's already resolved). To avoid unnecissary pauses, this check bypasses that.

Reply:
But this is an async function, so this will always have at least 1 tick delay. If you just return a promise resolved to the Response, we keep the same number of overall ticks and (slightly) simplify this method.